### PR TITLE
Hint and scope descriptor display fixes

### PR DESCRIPTION
**Pull request by gary-b**
1. Fix: When Batarang first opened the scopes would not show their friendly name.
2. Fix: After Batarang had loaded the hints tab would never update.

No tests for background.js as we would need to mock chrome's extension interface. https://github.com/vitalets/sinon-chrome might be an option.

Another PR request coming this weekend to fix a bug were scopes are not removed from the scope tree when the scope:destroy message is received.

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -31,18 +32,21 @@ function resetState(tabId) {
   };
 }
 
-function bufferData(tabId, message) {
-  var tabData = data[tabId],
-      scope;
-
+function setIsHintFlag(message) {
```
Comment (SomeKittens) [inline background.js]: For my own clarification, what does this accomplish over having the code in `bufferData`?

File: panel/components/inspected-app/inspected-app.js [hunk 2]
```diff
@@ -75,34 +75,34 @@ function inspectedAppService($rootScope, $q) {
...
   });
 
-  function onHintMessage(hint) {
-    if (hint.message) {
-      hints.push(hint);
-    } else if (hint.event) {
-      if (hint.event === 'hydrate') {
-        Object.keys(hint.data.scopes).forEach(function (scopeId) {
-          scopes[scopeId] = hint.data.scopes[scopeId];
+  function onHintMessage(message) {
```
Comment (SomeKittens) [inline panel/components/inspected-app/inspected-app.js]: I assume the reason for `hint` -> `message` here is to make the property `isHint` more sensible, correct?

File: background.js (hunk 1 above)
Comment (gary-b) [inline background.js]: It updates the message, which i wouldn't expect a function called bufferData to do.

File: panel/components/inspected-app/inspected-app.js (hunk 2 above)
Comment (gary-b) [inline panel/components/inspected-app/inspected-app.js]: Exactly.

**Comment by SomeKittens** [comment]
cherry-picked into master.  Thanks again, @gary-b!

**Pull request closed without merging**